hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/repair/om/OMRatisLogRepair.java

```diff
@@ -0,0 +1,300 @@
+            outputStream.write(newLogEntry);
+            outputStream.flush();
+          }
+          info("Replaced {" + oldRequest + "} with EchoRPC command at index "
+              + next.getIndex());
+        }
+      }
+
+      if (!isDryRun()) {
+        outputStream.flush();
+        outputStream.close();
+        Files.move(outputFile.toPath(), segmentFile.toPath(), StandardCopyOption.REPLACE_EXISTING);
+        info("Moved temporary output file to correct raft log location : " + segmentFile.toPath());
+      }
+
+    } catch (Exception ex) {
+      error("Exception: " + ex);
+    } finally {
+      if (logInputStream != null) {
+        logInputStream.close();
+      }
+      if (outputStream != null) {
+        outputStream.flush();
+        outputStream.close();
+      }
+      if (isDryRun()) {
+        boolean success = outputFile.delete();
+        if (!success) {
+          error("Error: Could not delete temporary output file \"" + outputFile + "\".");
+        }
+      }
+    }
+  }
+
+  private File createOutputFile(String name) throws IOException {
+    File temp = new File(name);
+    try {
+      if (temp.exists()) {
+        error("Warning: Temporary output file already exists - " + temp.getAbsolutePath() +
+            ". Trying to delete it and create a new one.");
+        boolean success = temp.delete();
+        if (!success) {
+          throw new IOException("Unable to delete old temporary file.");
+        }
+      }
+      boolean success = temp.createNewFile();
+      if (success) {
+        info("Temporary output file created successfully: " + temp.getAbsolutePath());
+      } else {
+        throw new IOException("createNewFile() failed.");
+      }
+    } catch (Exception e) {
+      throw new IOException("Error: Failed to create temporary output file - " + temp.getAbsolutePath(), e);
+    }
+    return temp;
+  }
+
+  private SegmentedRaftLogInputStream getInputStream(LogSegmentPath pi) {
+    try {
+      Class<?> logInputStreamClass =
+          Class.forName("org.apache.ratis.server.raftlog.segmented.SegmentedRaftLogInputStream");
+      Constructor<?> constructor = logInputStreamClass.getDeclaredConstructor(File.class, LogSegmentStartEnd.class,
+          SizeInBytes.class, SegmentedRaftLogMetrics.class);
+      constructor.setAccessible(true);
+      SegmentedRaftLogInputStream inputStream =
+          (SegmentedRaftLogInputStream) constructor.newInstance(segmentFile, pi.getStartEnd(),
+              SizeInBytes.valueOf("32MB"), null);
+      if (inputStream == null) {
+        throw new RuntimeException("logInputStream is null. Constructor might have failed.");
+      }
+      return inputStream;
+
+    } catch (ClassNotFoundException | NoSuchMethodException | SecurityException |
+             InvocationTargetException | InstantiationException | IllegalAccessException ex) {
+      error("Exception while trying to get input stream for segment file : " + ex);
+      throw new RuntimeException(ex);
```

Comment:
I think we only want to throw the exception, otherwise the error gets printed twice.